vtex-apps/pixel-manager
# Create new 'usePixelEventCallback' hook

#### What is the purpose of this pull request?

Create and export a hook capable responsible for registering event listeners to the `window`, with a certain `eventId` that will be sent through `message` events sent via `PixelContext.push`.

#### What problem is this solving?

This enables us to support custom user events.

#### How should this be manually tested?

Go to [Workspace](https://customevents--storecomponents.myvtex.com/) and add an item from the shelf to the minicart.

You should see two things happen:

1. There's a `modal-layout` above everything on the screen. This `modal-layout` triggered itself because it listens to a custom event sent by the `add-to-cart-button`. The `modal-layout` app is using the `usePixelEventCallback` hook implemented by this PR to listen to that event.
2. The minicart is now open. This is also due to the fact that the `drawer` used by the `minicart.v2` uses the `usePixelEventCallback` hook to listen to the same event being triggered by the `add-to-cart-button`.

#### Types of changes


#### How does this PR make you feel? [:link:](http://giphy.com/)

![](https://media.giphy.com/media/p3YOe3T0nZd1qgi81Y/giphy.gif)
